src/librustc_mir/monomorphize/polymorphize.rs

```diff
@@ -145,68 +135,27 @@ fn mark_used_by_predicates<'tcx>(
-                        debug!("mark_used_by_predicates: used!");
-                        for ty in trait_ref.substs.types() {
-                            mark_ty(unused_parameters, ty);
-                        }
-
-                        // No need to check for a type being used in the substs if `self_ty` was
-                        // used.
-                        continue;
-                    }
-
-                    // Consider `I` used if `T` is used in predicates of the form
-                    // `I: Iterator<Item = &'a (T, E)>` (see rust-lang/rust#75326)
-                    debug!("mark_used_by_predicates: checking substs");
-                    for ty in trait_ref.substs.types() {
-                        if is_ty_used(unused_parameters, ty) {
-                            debug!("mark_used_by_predicates: used!");
-                            mark_ty(unused_parameters, trait_ref.self_ty());
-                        }
-                    }
-                }
-                ty::PredicateAtom::Projection(proj, ..) => {
-                    let self_ty = proj.projection_ty.self_ty();
-                    debug!(
-                        "mark_used_by_predicates: (projection) self_ty={:?} proj.ty={:?}",
-                        self_ty, proj.ty
-                    );
-
-                    // Consider `T` used if `I` is used in predicates of the form
-                    // `<I as Iterator>::Item = T`
-                    debug!("mark_used_by_predicates: checking self");
-                    if is_ty_used(unused_parameters, self_ty) {
-                        debug!("mark_used_by_predicates: used!");
-                        mark_ty(unused_parameters, proj.ty);
-
-                        // No need to check for projection type being used if `self_ty` was used.
-                        continue;
-                    }
-
-                    // Consider `I` used if `T` is used in predicates of the form
-                    // `<I as Iterator>::Item = &'a (T, E)` (see rust-lang/rust#75326)
-                    debug!("mark_used_by_predicates: checking projection ty");
-                    if is_ty_used(unused_parameters, proj.ty) {
-                        debug!("mark_used_by_predicates: used!");
-                        mark_ty(unused_parameters, self_ty);
-                    }
-                }
+                ty::PredicateAtom::Trait(..) | ty::PredicateAtom::Projection(..) => (),
                 ty::PredicateAtom::RegionOutlives(..)
                 | ty::PredicateAtom::TypeOutlives(..)
                 | ty::PredicateAtom::WellFormed(..)
                 | ty::PredicateAtom::ObjectSafe(..)
                 | ty::PredicateAtom::ClosureKind(..)
                 | ty::PredicateAtom::Subtype(..)
                 | ty::PredicateAtom::ConstEvaluatable(..)
-                | ty::PredicateAtom::ConstEquate(..) => (),
+                | ty::PredicateAtom::ConstEquate(..) => continue,
```

Comment:
I wonder if we can just remove all this and ignore what kind of predicate it is.

Reply:
Removing this still passes all tests with polymorphization enabled, so pushed this change.